subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java

```diff
@@ -43,15 +46,17 @@ public interface ProjectLayout {
     DirectoryProperty getBuildDirectory();
 
     /**
-     * Creates a {@link RegularFile} provider whose location is calculated from the given {@link Provider}.
+     * {@inheritDoc}
```

Comment:
💅  As a user (who writes a build script and reads, at most, the javadoc of the method that I am calling, and not the javadoc of the class it belongs to), I would very much appreciate the methods javadoc to tell me relative to what directory the paths are resolved.

I would have definitely "overriden" the methods in both `ProjectLayout` and `SettingsLayout` just to clearly specify the reference location for the relative paths.

Reply:
Since that aspect applies to all methods mapping File/String paths (which are potentially relative) to file system locations (which are always absolute), I figured documenting it at the class level would make more sense.

Reply:
I am not saying we should remove the docs from the class level, but we should also add them to the method level. When it comes to the public API, we should be primarily thinking about the user convenience, rather than deduplication of the documentation.

I think our javadocs, especially historic ones, severely lack good content. We should not be chipping it away for the new APIs we introduce.